qt/scientific_interfaces/Indirect/test/IndirectFitDataTest.h

```diff
@@ -0,0 +1,447 @@
+#ifndef MANTID_INDIRECTFITDATATEST_H_
+#define MANTID_INDIRECTFITDATATEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "IndirectFitData.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+#include <iostream>
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace MantidQt::CustomInterfaces::IDA;
+
+namespace {
+
+IndirectFitData getIndirectFitData(std::size_t const &numberOfSpectra,
+                                   std::size_t const &numberOfBins) {
+  auto const workspace = WorkspaceCreationHelper::create2DWorkspace123(
+      numberOfSpectra, numberOfBins);
+  Spectra const spec = std::make_pair(0u, workspace->getNumberHistograms() - 1);
+  IndirectFitData data(workspace, spec);
+  return data;
+}
+
+/// Simple class to set up the ADS with the configuration required
+struct SetUpADSWithWorkspace {
+
+  SetUpADSWithWorkspace(std::string const &inputWSName,
+                        IndirectFitData const &data) {
+    AnalysisDataService::Instance().addOrReplace(inputWSName, data.workspace());
+  };
+
+  ~SetUpADSWithWorkspace() { AnalysisDataService::Instance().clear(); };
+};
+
+/// This is used to compare Spectra which is implemented as a boost::variant
+struct AreSpectraEqual : public boost::static_visitor<bool> {
+
+  template <typename T, typename U>
+  bool operator()(const T &, const U &) const {
+    return false; // cannot compare different types
+  }
+
+  template <typename T> bool operator()(const T &lhs, const T &rhs) const {
+    return lhs == rhs;
+  }
+};
+
+} // namespace
+
+class IndirectFitDataTest : public CxxTest::TestSuite {
+public:
+  static IndirectFitDataTest *createSuite() {
+    return new IndirectFitDataTest();
+  }
+
+  static void destroySuite(IndirectFitDataTest *suite) { delete suite; }
+
+  void test_data_is_instantiated_correctly() {
+    auto const workspace = WorkspaceCreationHelper::create2DWorkspace123(1, 3);
+    Spectra const spec =
+        std::make_pair(0u, workspace->getNumberHistograms() - 1);
+
+    workspace->setTitle("Test Title");
+    IndirectFitData const data(workspace, spec);
+
+    TS_ASSERT_EQUALS(data.workspace(), workspace);
+    TS_ASSERT_EQUALS(data.workspace()->getTitle(), "Test Title");
+    TS_ASSERT_EQUALS(data.workspace()->getNumberHistograms(), 1);
+  }
+
+  void test_that_DiscontinuousSpectra_is_set_correctly() {
```

Comment:
Is it worth extending this test to cover some extra functionality? For example

- What if I supply numbers in the wrong order?
- What if I supply a string with duplicates in?
- What about typos like "1,-2"; is this a negative number, or badly formatted range?

Reply:
The string passed to create a DiscontinuousSpectra like in `DiscontinuousSpectra<std::size_t> const spec(string);` is validated before getting to the point of its creation, and so I don't believe a test on this is required?

Reply:
I am adding three new tests to test this so thanks for spotting it!